### ISSUE TITLE
Improve error message on invalid YAML 'pattern'

### ISSUE THREAD

**Issue by mschwager** [returntocorp/sgrep#210]
I accidentally used `pattern` instead of `patterns` here:

```yaml
rules:
  - id: test-pattern
    pattern:
        - pattern: "$D = {}"
    message: "test"
    languages: [python]
    severity: WARNING
```

Which gives the following error:

```
Traceback (most recent call last):
  File "/bin/sgrep-lint", line 1119, in <module>
    main(args)
  File "/bin/sgrep-lint", line 903, in main
    invalid_patterns = validate_patterns(valid_configs)
  File "/bin/sgrep-lint", line 649, in validate_patterns
    patterns_with_ids = list(
  File "/bin/sgrep-lint", line 144, in enumerate_patterns_in_boolean_expression
    yield from enumerate_patterns_in_boolean_expression(pattern_or_list[2])
  File "/bin/sgrep-lint", line 140, in enumerate_patterns_in_boolean_expression
    if isinstance(pattern_or_list[2], list):
KeyError: 2
```

Switching to `patterns` fixes the issue:

```yaml
rules:
  - id: test-pattern
    patterns:
        - pattern: "$D = {}"
    message: "test"
    languages: [python]
    severity: WARNING
```

It's difficult to determine what's wrong with the YAML file initially. Let's help our users out by giving them a helpful error message :+1:

**Comment by ievans** [comment]
Agreed, any `patteren-X` that can have leaf nodes ought to be valid there.

**Comment by ievans** [comment]
Fixed in #198
